US-CBP/cbp-theme
# Ds css build - Style Linting && Husky setup

What's this PR do?

Mainly this is for setting up the initial style linting processes, with some additional things like:

    added a "stylelint-README.md" file
    Created a "/tests/" folder for saving any errors produced by the style linting within a file called "tests/linting/stylelint-errors.log"
    created a "CSS-Naming-System.txt" file for placing our rules within
    added the "LICENSE.MD" file

Where should the reviewer start?

First should glance at: style linting rules within the .stylelintrc && added options added to the stylelinting plugin w/in the 'webpack.config.js' file && the 'stylelint-README.md' file

Then, you will want to purposely create some styling 'errors' for the linting to find and show how it handles everything. So maybe go to one of the .scss files within /styles/** and add some random rules and comments to use for testing the actual process.
(Because the current .scss files should all be meeting the linting requirements set)
How should this be manually tested?

Next, once you think you have some new test code to produce some errors,

can run a few different scripts:

    npm run lint:scss

    npm run build

    npm run lint
    (They are all running the same style linting scripts)

Next, after you fix the listed 'errors', go ahead and get the index.html page running

    npm run local

--> then place some more bad sass code within one of the .scss files and see how the live page shows the errors. (** It should show an overlay for listing them **)

Next, after testing all the above processes, can test the Husky hooks for commit and push.
Any background context you want to provide?

n/a
What are the relevant tickets?

n/a
Screenshots (if appropriate)

n/a
Questions:

    Is there a blog post?

n/a

    Does the knowledge base need an update?

n/a
